File: 04 - Binary Trees/01 - Constructor/BinaryTree.cs
public class BinaryTree<T>
{
    public class Node
    {
        // The Node needs a value and pointers to Left and Right.
        public T Value { get; set; }
        public Node? Left { get; set; }
        public Node? Right { get; set; }
        public Node(T value) { this.Value = value; Left = null; Right = null; }
    }
    // A Binary Tree needs a "root".
    public Node? Root { get; private set; }
    public BinaryTree()
    {
        // We won't create an initial Node. The root will be 'null'.
        this.Root = null;
    }
}

File: 04 - Binary Trees/01 - Constructor/BinaryTree.py
from __future__ import annotations
from typing import Any


class Node:
    def __init__(self: Node, value: Any) -> None:
        self.value: Any = value
        self.left: Node | None = None
        self.right: Node | None = None


class BinaryTree:
    def __init__(self: BinaryTree) -> None:
        # We won't create an initial Node. The root will be None.
        self.root: Node | None = None


my_bt: BinaryTree = BinaryTree()

print(my_bt.root)  # None
